### ISSUE TITLE
Sketchup::Text  #leader_type= not setting properly sometimes

### ISSUE THREAD

**Issue by AntonCha** [SketchUp/api-issue-tracker#792]
1. SketchUp/LayOut Version: Sketchup Pro 2022
2. OS Platform: Mac - M1 machine

when setting the #leader_type= of a Sketchup::Text object sometimes the value is not set (possible asynchronous problem)

The result is that sometimes the leader_type still has value 2, although the code is setting it to 1

```ruby
module Testing

  def self.go
    model = Sketchup.active_model
    inst  = model.selection.first
    outer_entities = inst.definition.entities
    types = []
    # 
    model.start_operation("Text Test",true)
    group = outer_entities.add_group
    group.locked = true
    inner_entities = group.entities
    cpt  = inner_entities.add_cpoint(ORIGIN)
    text = inner_entities.add_text(
      "TestName",
      Geom::Point3d.new(30, 30, 30),
      Geom::Vector3d.new(0, 0, 35)
    )
    text.leader_type = 1
    text.arrow_type  = 2
    text.display_leader = true
    types << text.leader_type
    cpt.erase!
   
    puts "True leader types: #{types.inspect}" if text.leader_type == 2
    model.commit_operation
  end # go()

end # module

i = 0
timer = UI.start_timer(0.01, true) { 
  Testing.go
  i += 1
  UI.stop_timer(timer) if i > 1000
}
```

when a component is selected in the scene and the code gets executed (notice i call Testing.go 1000 times) i get occasions where the value is not set properly)

**Comment by DanRathbun** [comment]
The whole point of collecting the leader types into a `types` array is that this is kept **outside** the test loop and displayed 
**afterward**. You've rewritten my test method edition and have the array inside the test loop so it always has only 1 member.

REF: https://forums.sketchup.com/t/applying-add-text-and-leader-type-multiple-times-yields-different-parameter-values/201776/2

Another way to test if the objects themselves are truly having their leader types set correctly is to collect the text callout object references into the array during the loop, and then **afterward** iterate the array and map it to leader types.

Also, your edition will create 1000 undo operations the way you show running the test. (Not efficient.)

In this edition I've added a `num` argument to the method so it can be called with the number of times you wish the inner loop to run. But, all the object creation still happens within **one** undo operation. (Note that when an issue is reported I name the test module after the issue id number ... `Issue792`.)
```ruby
module Issue792

  def self.go(num)
    model = Sketchup.active_model
    inst  = model.selection.first
    outer_entities = inst.definition.entities
    objects = []
    model.start_operation("Text Test",true)
    #
    num.times do
      group = outer_entities.add_group
      group.locked = true
      inner_entities = group.entities
      cpt  = inner_entities.add_cpoint(ORIGIN)
      text = inner_entities.add_text(
        "TestName",
        Geom::Point3d.new(30, 30, 30),
        Geom::Vector3d.new(0, 0, 35)
      )
      text.leader_type = 1
      text.arrow_type  = 2
      text.display_leader = true
      puts text.leader_type # slow IO
      objects << text
      cpt.erase!
    end
    #
    model.commit_operation
    types = objects.map(&:leader_type)
    puts "True leader types: #{types.inspect}"
  end # go()

end # module
```

----

Also, whenever strange things happen, I always suggest rebooting the machine to clear out memory, and retesting.

**Comment by DanRathbun** [comment]
What is the point of ...

1. Locking the groups ?

2. Calling the test method from within a timer (when core Ruby iterative constructs might suffice) ?

**Comment by AntonCha** [comment]
1. locking the groups makes no difference, i was just experimenting
2. calling the method from within a timer is just used for duplicating the bug easier for you.

Calling the method 1000 time is just an over exaggeration to make sure you also get the undefined behaviour as i do. I am getting this problem with calling Testing.go only 10-12 times, but i wanted to make sure you also can reproduce it.

I used your code, because your point was that `puts` might be slow, while adding the object inside an array would not have that behaviour. My edited version of the code uses your suggestion and still reproduces the issue as well as shows that puts is not the problem in the matter, because when puts has wrong value, the array also has wrong value inside of it.

The problem that i am representing might not come from the add_text method, but from other (previous) part inside that Testing.go function. I have noticed the bug thanks to that method, but i cant really go too much deeper into the actual code to investigate myself. Thats why i tried to give you a code snippet that should trigger the problem with high possibility(that is where the timer with 1000 calls comes in) and to at least point your attention to the direction of the problem.

**Comment by DanRathbun** [comment]
Okay, but if you do not call it from within a timer, do you you still get the errors? Ie, it might be a timer issue.

**Comment by AntonCha** [comment]
yes, i do get the errors without the timer. The timer was just to help illustrate the problem better without a lot of preparation on the side of the testing developer

**Comment by DanRathbun** [comment]
Is this an Intel Mac or a new M1 silicon machine ?

**Comment by kalpana-ghodeswar** [comment]
Closing this as not reproducible on SU2025.0
